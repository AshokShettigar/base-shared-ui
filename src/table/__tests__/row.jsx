import { shallow } from 'enzyme';
import Cell from 'platform-common-ui/table/cell';
import Row from 'platform-common-ui/table/row';

describe('table/row', function() {
    it('should render the row', function() {
        const wrapper = shallow(
            <Row>
                <Cell>Data 1</Cell>
                <Cell>Data 2</Cell>
            </Row>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should render the row with isSelected', function() {
        const wrapper = shallow(
            <Row isSelected>
                <Cell>Data 1</Cell>
                <Cell>Data 2</Cell>
            </Row>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should render the row with class name', function() {
        const wrapper = shallow(
            <Row className="my-class">
                <Cell>Data 1</Cell>
                <Cell>Data 2</Cell>
            </Row>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should test onClick on row', function(done) {
        const onClickHandler = () => {
            expect(true).toBeTruthy();
            done();
        };

        const wrapper = shallow(
            <Row onClick={onClickHandler}>
                <Cell>Data 1</Cell>
                <Cell>Data 2</Cell>
            </Row>
        );
        wrapper.simulate('click');
    });
});
