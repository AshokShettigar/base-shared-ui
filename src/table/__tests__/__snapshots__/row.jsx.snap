// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`table/row should render the row 1`] = `
<tr
  className="cn-table__row"
  data-component="Row"
>
  <TableCell>
    Data 1
  </TableCell>
  <TableCell>
    Data 2
  </TableCell>
</tr>
`;

exports[`table/row should render the row with class name 1`] = `
<tr
  className="cn-table__row my-class"
  data-component="Row"
>
  <TableCell>
    Data 1
  </TableCell>
  <TableCell>
    Data 2
  </TableCell>
</tr>
`;

exports[`table/row should render the row with isSelected 1`] = `
<tr
  className="cn-table__row cn-table__row--selected"
  data-component="Row"
>
  <TableCell>
    Data 1
  </TableCell>
  <TableCell>
    Data 2
  </TableCell>
</tr>
`;
